File: ios/SFPYCardCaptureManager.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(SFPYCardCapture, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(onReady, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onCardChange, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onValidated, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onError, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onProceedToAuthentication, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(environment, NSString)
RCT_EXPORT_VIEW_PROPERTY(authToken, NSString)
RCT_EXPORT_VIEW_PROPERTY(tracker, NSString)
RCT_EXPORT_VIEW_PROPERTY(disabled, BOOL)
RCT_EXPORT_VIEW_PROPERTY(autofocus, BOOL)
RCT_EXTERN_METHOD(submit:(nonnull NSNumber*) reactTag)
RCT_EXTERN_METHOD(clear:(nonnull NSNumber*) reactTag)
@end

File: ios/SFPYCardCaptureManager.swift
import Foundation
import React

@objc(SFPYCardCapture)
class SFPYCardCaptureManager: RCTViewManager {
    override func view() -> UIView! {
        return SFPYCardCaptureView()
    }

    @objc func submit(_ reactTag: NSNumber) {
        bridge?.uiManager.addUIBlock { _, viewRegistry in
            if let view = viewRegistry?[reactTag] as? SFPYCardCaptureView {
                view.submit()
            }
        }
    }

    @objc func clear(_ reactTag: NSNumber) {
        bridge?.uiManager.addUIBlock { _, viewRegistry in
            if let view = viewRegistry?[reactTag] as? SFPYCardCaptureView {
                view.clear()
            }
        }
    }

    override static func requiresMainQueueSetup() -> Bool {
        return false
    }
}
